examples/chart/teleport-cluster/templates/proxy/deployment.yaml

```diff
@@ -161,8 +170,17 @@ spec:
           periodSeconds: 5 # poll health every 5s
           failureThreshold: 12 # consider agent unhealthy after 60s (12 * 5s)
           timeoutSeconds: {{ $proxy.probeTimeoutSeconds }}
-{{- if $proxy.postStart.command }}
         lifecycle:
+          # waiting during preStop ensures no new request will hit the Terminating pod
+          # on clusters using kube-proxy (kube-proxy syncs the node iptables rules every 30s)
+          preStop:
```

Comment:
I wonder if we could also use this to have proxies unregister themselves on shutdown in future to mitigate https://github.com/gravitational/teleport/issues/20057

Reply:
Unregistering would be a nice improvement, but I'm not sure `preStop` is the best place because it is implementation-specific (Terraform/CloudFormation have the same issue), and we don't have control over the concurrency with `preStop`. Maybe the proxy will take 2 minutes to shut down, and I don't know what would happen if it gets unregistered while it has not finished doing its job. IMO the best place to do it would be in the graceful shutdown sequence.

Agendas are a little packed right now because of v12, but I'll raise this issue to folks who know more about registration and shutdown sequence than me once things settle down.